File: src/test/java/info/skyblond/archivedag/controller/grpc/GrpcTestUtils.java
package info.skyblond.archivedag.controller.grpc;

import info.skyblond.archivedag.protos.common.Empty;
import io.grpc.internal.testing.StreamRecorder;
import org.junit.jupiter.api.function.Executable;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

public class GrpcTestUtils {
    public static void checkEmptyResponse(StreamRecorder<Empty> responseObserver) {
        assertNull(responseObserver.getError());
        List<Empty> results = responseObserver.getValues();
        assertEquals(1, results.size());
        assertEquals(Empty.getDefaultInstance(), results.get(0));
    }

    public static void safeExecutable(Executable executable) {
        try {
            executable.execute();
        } catch (Throwable ignored) {
        }
    }
}

File: backend/src/test/kotlin/info/skyblond/archivedag/arudaz/controller/grpc/GrpcTestUtils.kt
package info.skyblond.archivedag.arudaz.controller.grpc

import info.skyblond.archivedag.arudaz.protos.common.Empty
import io.grpc.internal.testing.StreamRecorder
import org.junit.jupiter.api.Assertions

fun checkEmptyResponse(responseObserver: StreamRecorder<Empty>) {
    Assertions.assertNull(responseObserver.error)
    val results = responseObserver.values
    Assertions.assertEquals(1, results.size)
    Assertions.assertEquals(Empty.getDefaultInstance(), results[0])
}
